#! /bin/sh /usr/share/dpatch/dpatch-run
## 12_check_ssh_read_socket.dpatch by Jan Wagner <waja@cyconet.org>
##
## From a3871201821d9abe8a12e637e7dd00d9d1dde2a5 Mon Sep 17 00:00:00 2001
## From: Jan Wagner <waja@cyconet.org>
## Date: Fri, 10 Jan 2014 15:56:30 +0100
## Subject: [PATCH] check_ssh: Get rid of sshd: Read from socket failed:
##  Connection reset by peer
##
## This fix was grabbed from FreeBSD downstream and provided by Dmitry Sivachenko.
## Fixes Debian Bug #734811

@DPATCH@

---
 plugins/check_ssh.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plugins/check_ssh.c b/plugins/check_ssh.c
index 1c032a4..2d63528 100644
--- a/plugins/check_ssh.c
+++ b/plugins/check_ssh.c
@@ -250,6 +250,7 @@
 			printf
 				(_("SSH WARNING - %s (protocol %s) version mismatch, expected '%s'\n"),
 				 ssh_server, ssh_proto, remote_version);
+			recv (sd, output, BUFF_SZ, 0);
 			close(sd);
 			exit (STATE_WARNING);
 		}
@@ -260,6 +261,7 @@
 			(_("SSH OK - %s (protocol %s) | %s\n"),
 			 ssh_server, ssh_proto, fperfdata("time", elapsed_time, "s",
 			 FALSE, 0, FALSE, 0, TRUE, 0, TRUE, (int)socket_timeout));
+		recv (sd, output, BUFF_SZ, 0);
 		close(sd);
 		exit (STATE_OK);
 	}
-- 
1.8.5.1

